### PR TITLE
[MaterializedView]Partition Prune support expr

### PR DESCRIPTION
**Pull request by zddr**
## What type of PR is this：
- [ ] bug
- [x] feature
- [ ] enhancement
- [ ] refactor
- [ ] others

## Which issues of this PR fixes ：

Fixes #

### CODE REVIEW

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage check]

:disappointed: **fail** : 1 / 2 (50.00%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/optimizer/operator/ColumnFilterConverter.java|1|2|50.00%|

**Comment by Astralidea** [comment]
src/main/java/com/starrocks/sql/optimizer/operator/ColumnFilterConverter.java:38:8: Unused import - org.antlr.v4.runtime.atn.LexerPushModeAction. [UnusedImports]

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage check]

:heart_eyes: **pass** : 45 / 53 (84.91%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/optimizer/operator/ColumnFilterConverter.java|43|51|84.31%|
|:large_blue_circle:|com/starrocks/sql/optimizer/operator/logical/LogicalScanOperator.java|1|1|100.00%|
|:large_blue_circle:|com/starrocks/sql/optimizer/rule/transformation/PartitionPredicatePrune.java|1|1|100.00%|

File: fe/fe-core/src/main/java/com/starrocks/sql/optimizer/operator/ColumnFilterConverter.java [hunk 1]
```diff
@@ -80,12 +97,92 @@ private static boolean checkColumnRefCanPartition(ScalarOperator right) {
                 return type == columnType || (type != Type.LARGEINT && columnType != Type.LARGEINT);
             }
 
+            if (type.isDateType() && columnType.isDateType()) {
```
Comment (kangkaisen) [inline fe/fe-core/src/main/java/com/starrocks/sql/optimizer/operator/ColumnFilterConverter.java:100]: if one type is date, another type is datetime, I think there will be some issues.

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage check]

:heart_eyes: **pass** : 44 / 51 (86.27%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/optimizer/operator/ColumnFilterConverter.java|42|49|85.71%|
|:large_blue_circle:|com/starrocks/sql/optimizer/operator/logical/LogicalScanOperator.java|1|1|100.00%|
|:large_blue_circle:|com/starrocks/sql/optimizer/rule/transformation/PartitionPredicatePrune.java|1|1|100.00%|

**Comment by ABingHuang** [comment]
run starrocks_fe_unittest

**Comment by wangruin** [comment]
run starrocks_fe_unittest

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage check]

:heart_eyes: **pass** : 45 / 51 (88.24%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/optimizer/operator/ColumnFilterConverter.java|43|49|87.76%|
|:large_blue_circle:|com/starrocks/sql/optimizer/operator/logical/LogicalScanOperator.java|1|1|100.00%|
|:large_blue_circle:|com/starrocks/sql/optimizer/rule/transformation/PartitionPredicatePrune.java|1|1|100.00%|

**Review by kangkaisen — APPROVED** [review]

**Review by ABingHuang — APPROVED** [review]
LGTM

**Review by Astralidea — APPROVED** [review]

**Pull request merged**